canonical-web-and-design/maas-ui
# feat(ui): add ability to delete domain records

## Done

- Added delete actions to domain slice
- Fixed records table so that it works with records that have the same `dnsresource_id`
- Added deleteDomainRecord saga for handling whether to delete the underlying DNS resource if it's the last record in that resource
- Fixed the updateDomainRecord saga so that it works with records that share a `dnsresource_id`

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/domains and add a new domain
- Add an A record with three ips, e.g. "1.1.1.1,2.2.2.2,3.3.3.3"
- Open the edit form for the first A record and change the Name and Data fields and click submit
- The A record should change and the other two should remain unchanged
- Delete the A record that you changed.
- Actions should be dispatched for `deleteAddressRecord` and `deleteDNSResource`
- Delete one of the other records, which should share a name
- Only the action to `deleteAddressRecord` should be dispatched
- Delete the last A record
- Add two TXT records that have the same name
- Check that you can edit each record independently
- Delete the first TXT record
- Only an action to `deleteDNSData` should be dispatched
- Delete the other TXT record
- Actions should be dispatched for `deleteDNSData` and `deleteDNSResource`
- Delete the domain - you shouldn't get any errors about not being able to delete the domain because of some leftover DNS resources

## Fixes

Fixes canonical-web-and-design/app-squad#52
